odoo/models.py

```diff
@@ -2904,6 +2872,8 @@ def _add_sql_constraints(self):
     def _execute_sql(self):
         """ Execute the SQL code from the _sql attribute (if any)."""
         if hasattr(self, "_sql"):
+            warnings.warn(f'_sql attribute is depreciated ({self._name}), override _auto_init instead',
+                          DeprecationWarning)
```

Comment:
I wouldn't use a warnings.warn here as it would log the entire (very long) callstack which is useless here. Better use `logging.warn(exc_info=DeprecationWarning(""))` here

Reply:
Or I can add `stacklevel=2` (same than in other place) to avoid a long callstack and be consistent ?

Reply:
They're completely different things, `stacklevel=2` is used to report a location higher up the stack as the source of the issue, the stacktrace comes from the override between the warnings and logging system (because when hunting down DeprecationWarnings it was hell when they came from deep inside a dependency).

Reply:
Ok, I get it. I misunderstand the argument. The `stacklevel=2` is still pertinent to avoid showing the code of the  warning it -self. Thank you.